checker/src/main/java/org/checkerframework/checker/determinism/DeterminismVisitor.java

```diff
@@ -488,40 +494,44 @@ public Void visitMethod(MethodTree node, Void p) {
                             .getAnnotationInHierarchy(atypeFactory.NONDET));
         }
         boolean isPolyPresent = false;
-        boolean isPolyUpPresent = false;
-        boolean isPolyDownPresent = false;
         boolean isPolyUsePresent = false;
-        boolean isPolyUpDetPresent = false;
         for (AnnotationMirror atm : polyAnnotations) {
             if (AnnotationUtils.areSame(atm, atypeFactory.POLYDET_UP)) {
```

Comment:
The condition expression of every if statement is executed even if the first one is true.  Also, it would be more efficient to check if the annotation is `@PolyDet` with any argument, `AnnotationUtils#areSameByClass`, then check the argument using `AnnotationUtils#getElementValue`.